R/tm_outliers.R

```diff
@@ -241,7 +241,41 @@ srv_outliers <- function(id, data, reporter, filter_panel_api, outlier_var,
   checkmate::assert_class(data, "tdata")
   moduleServer(id, function(input, output, session) {
     vars <- list(outlier_var = outlier_var, categorical_var = categorical_var)
-    selector_list <- teal.transform::data_extract_multiple_srv(vars, data)
+
+    selector_list <- teal.transform::data_extract_multiple_srv(
+      data_extract = vars,
+      datasets = data,
+      select_validation_rule = list(
+        outlier_var = shinyvalidate::compose_rules(
+          shinyvalidate::sv_required("Please select a variable"),
+          ~ if (
+            length(selector_list()$categorical_var()$select) > 0 &&
+              selector_list()$categorical_var()$select == (.)) {
+            "`Variable` and `Categorical factor` cannot be the same"
+          }
+        )
+      ),
+      filter_validation_rule = list(
```

Comment:
these should involve `(.)` ?

Also if app developer does not include the categorical_var argument then selector_list()$categorical_var is NULL 
![image](https://user-images.githubusercontent.com/15201933/209333324-018da3ff-4f70-41a3-8a38-05ea65dc5f91.png)

Reply:
updated to account for this.